### PR TITLE
Add lru cache

### PR DESCRIPTION
**Pull request by godchen0212**
Add lru cache
Issue #15540

Signed-off-by: godchen0212 <qingxiang.chen@zilliz.com>

### CODE REVIEW

File: internal/storage/lru_cache.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
+// Licensed to the LF AI & Data foundation under one
```
Comment (xiaofan-luan) [inline internal/storage/lru_cache.go:1]: what is the difference between our cache and  https://github.com/hashicorp/golang-lru/blob/80c98217689d6df152309d574ccc682b21dc802c/simplelru/lru.go?

**Review by xiaofan-luan — COMMENTED** [review]
also I believe lru.go  should be a better fit since it is thread safe

**Comment by godchen0212** [comment]
Their lru.go does not implement thread safe. If we implement lru.go in our own package, we can easily use private variables to combine chunk manager, and then customize and upgrade according to our demands.

**Pull request merged**